simple-markdown.js

```diff
@@ -753,7 +753,7 @@ var TABLES = (function() {
 
 var LINK_INSIDE = "(?:\\[[^\\]]*\\]|[^\\[\\]]|\\](?=[^\\[]*\\]))*";
 var LINK_HREF_AND_TITLE =
-        "\\s*<?((?:[^\\s\\\\]|\\\\.)*?)>?(?:\\s+['\"]([\\s\\S]*?)['\"])?\\s*";
+        "\\s*<?((?:\\([^)]*\\)|[^\\s\\\\]|\\\\.)*?)>?(?:\\s+['\"]([\\s\\S]*?)['\"])?\\s*";
```

Comment:
Two minor things I think this might not handle yet are:

1. It allows unescaped opening quotes in the match, which results in confusing matches, such as: `[link](http://example.com/test((hi))` rendering to `<a href="http://example.com/test((hi)">link</a>`.
    * I think we might be able to just add `(` to the list of forbidden characters in the parenthesis capture like `\\([^()]*\\)` to avoid this.

2. Escaping the closing `)`, like: `[link](http://example.com/test(hi\))`. It's... pretty unlikely that anyone really needs that 😄, but we do support it in other places.
    * I think something like: `\\((?:[^)\\\\]|\\\\.)*\\)` instead of `\\([^)]*\\)` might let that work (see how the later block uses `[^\\s\\\\]|\\\\.` to do the same thing).